scripts/check_go_version.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+MOD_VERSION=$(go mod edit -json | jq -r .Go)
+echo "go.mod version: $MOD_VERSION"
+
+for wf in .github/workflows/*.yml; do
+    echo "Checking $wf ..."
+
+    WF_VERSIONS=$(yq -r '.jobs[].steps[] | select(.with["go-version"]) | .with["go-version"]' -o=tsv "$wf" | grep -v '^---$' || true)
+    if [ -z "$WF_VERSIONS" ]; then
```

Comment:
```suggestion
    if [[ -z "$WF_VERSIONS" ]]; then
```
We're using Bash, not sh 😄